src/librustc/mir/repr.rs

```diff
@@ -373,8 +381,11 @@ impl<'tcx> TerminatorKind<'tcx> {
                 slice::ref_slice(t).into_cow(),
             Call { destination: None, cleanup: Some(ref c), .. } => slice::ref_slice(c).into_cow(),
             Call { destination: None, cleanup: None, .. } => (&[]).into_cow(),
-            Drop { target, unwind: Some(unwind), .. } => vec![target, unwind].into_cow(),
-            Drop { ref target, .. } => slice::ref_slice(target).into_cow(),
+            DropAndReplace { target, unwind: Some(unwind), .. } |
+            Drop { target, unwind: Some(unwind), .. } =>
+                vec![target, unwind].into_cow(),
+            DropAndReplace { ref target, .. } | Drop { ref target, .. } =>
```

Comment:
Nit: can you write `unwind: None` here to make this more robust against accidental re-ordering? (and more obvious, I missed the lines above for a while)